core/modules/user/user.admin.inc

```diff
@@ -684,6 +691,7 @@ function user_admin_role_submit($form, &$form_state) {
     );
     user_role_save($role);
     backdrop_set_message(t('The %role role has been added.', array('%role' => $role->label)));
+    $redirect = 'admin/config/people/permissions';
```

Comment:
There is no logic to prevent the form going here if the plain 'save role' btn is clicked.

Reply:
Thank you @docwilmot!! ...how could I miss that 🤦🏼 

Fixed with: https://github.com/klonos/backdrop/commit/a3d67674445a3c45fd4219a2c0d1ab431ae7bf87